usecases/internal/measurement.go

```diff
@@ -28,11 +30,19 @@ func MeasurementPhaseSpecificDataForFilter(
 		return nil, api.ErrDataNotAvailable
 	}
 
+	// Validate validator parameter
+	if validator == nil {
+		return nil, fmt.Errorf("validator is required")
+	}
+
 	var result []float64
 
 	for _, item := range data {
-		if item.Value == nil || item.MeasurementId == nil {
-			continue
+		// Use validator instead of basic nil checks
+		if err := validator.Validate(&item); err != nil {
+			// For MGCP-003 compliance, we explicitly handle ErrSkipMeasurement
+			// All other validation errors also result in skipping the measurement
+			continue // Skip invalid measurements, don't fail entire operation
```

Comment:
Although the reasoning is correct, this error message is confusing as no error is handled explicitly.
It might be nice to have some way of logging/warning/reporting measurements with valueState != normal and probably another one for other validationErrors, but I don't know if we need that in this PR.
```suggestion
			// Due to MGCP-003, and MPC-003 we want to skip measurements with valueState != normal
			// We also want to skip all other validation errors so that we don't fail the entire operation
			// just because the remote contains invalid data
			continue
```